include/swift/Runtime/Concurrent.h

```diff
@@ -601,7 +601,8 @@ template <class ElemTy> struct ConcurrentReadableHashMap {
   /// Otherwise, just return the passed-in size, which is always valid even if
   /// not necessarily optimal.
   size_t goodSize(size_t size) {
-#if defined(__APPLE__) && defined(__MACH__)
+#if defined(__APPLE__) && defined(__MACH__) && \
+        defined(SWIFT_STDLIB_HAS_DARWIN_LIBMALLOC)
```

Comment:
You'll want this to just be `SWIFT_STDLIB_HAS_DARWIN_LIBMALLOC` (no `defined()` around it) to handle the case where it's `0`.